client/src/drawing/index.js

```diff
@@ -97,162 +86,171 @@ export const draw = () => {
     }
   }
   //draw game objects
-  for (const go in sp.gameObjects) {
-    sp.gameObjects[go].draw();
+  for (const go in sg.pickups) {//TODO: here
+    spriteRenderer.drawPickup(s, sg.pickups[go]);
+  }
+  for (const go in sg.players) {
+    playerRenderer.draw(s, sg.players[go]);
   }
 }
 
-export const drawGUI = () => {
+export const drawGUI = (s) => {
+  let sg = s.game;
   if (sg.clientState === CLIENT_STATES.PLAYING || sg.clientState === CLIENT_STATES.PAUSED);
-  switch (sg.gameState) {
+  switch (sg.gameState) {//TODO: there is a minor sync bug here relating to the gameState being updated without being sent the relevant data
     case CLIENT_STATES.GAME_WAITING_FOR_PLAYERS:
-      drawWaitingForPlayersGUI();
-      break;
+    {//drawWaitingForPlayersGUI
```

Comment:
should probably break these back out into their own functions so its easier to read :)